src/lib.rs

```diff
@@ -3427,6 +3429,74 @@ impl Build {
             }
         }
 
+        let deployment_from_env = |name: &str| {
+            // note this isn't hit in production codepaths, its mostly just for tests which don't
+            // set the real env
+            if let Some((_, v)) = self.env.iter().find(|(k, _)| &**k == OsStr::new(name)) {
+                Some(v.to_str().unwrap().to_string())
+            } else {
+                env::var(name).ok()
+            }
+        };
+
+        // Determines if the acquired deployment target is too low to support modern C++ on some Apple platform.
+        //
+        // A long time ago they used libstdc++, but since macOS 10.9 and iOS 7 libc++ has been the library the SDKs provide to link against.
+        // If a `cc`` config wants to use C++, we round up to these versions as the baseline.
+        let maybe_cpp_version_baseline = |deployment_target_ver: String| -> String {
+            if !self.cpp {
+                return deployment_target_ver;
+            }
+
+            let mut deployment_target = deployment_target_ver
+                .split('.')
+                .map(|v| v.parse::<u32>().expect("integer version"));
+
+            match os {
+                AppleOs::MacOs => {
+                    // Targeting higher then 10.x (11.0+)?
+                    let major_supported = deployment_target
+                        .next()
+                        .map(|major| major > 10)
+                        .unwrap_or_default();
```

Comment:
```suggestion
                    let major = deployment_target.next().unwrap_or(0);
```

I think it's more readable to obtain the major and minor, then test it in `if`.